tap_appstore/client.py

```diff
@@ -18,12 +23,112 @@
     """AppStore stream class."""
+    DATE_FORMAT = '%Y-%m-%d'
+    DATE_INCREMENT = timedelta(days=1)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.api = self.setup_api_connection()
+
+        date_format_mapping = {
+            'begin_date': '%m/%d/%Y',
+            'start_date': '%m/%d/%Y',
+            'end_date': '%m/%d/%Y',
+            '_api_report_date': '%Y-%m-%d',
+        }
+        self.date_fields = {name: date_format_mapping.get(name, '%Y-%m-%d') for name, prop in
+                            self.schema["properties"].items() if
+                            'string' in prop['type'] and prop.get('format') == 'date-time'}
+        self.float_fields = {name for name, prop in self.schema["properties"].items() if 'number' in prop['type']}
+        self.int_fields = {name for name, prop in self.schema["properties"].items() if 'integer' in prop['type']}
+
+    def setup_api_connection(self):
+        """Set up the API connection using provided configuration."""
+        return Api(self.config['key_id'], self.config['key_file'], self.config['issuer_id'], submit_stats=False)
+
+    def download_data(self, start_date, api):
+        """Set up the endpoint for the API call. Override in subclass as needed."""
+        raise NotImplementedError("Subclasses must implement this method.")
+
+    def process_record(self, record):
+        """Convert ints, floats, and date fields"""
+        if self.float_fields:
+            self.convert_fields(record, self.float_fields, float)
+
+        if self.int_fields:
+            self.convert_fields(record, self.int_fields, int)
+
+        if self.date_fields:
+            for field, format in self.date_fields.items():
+                if field in record and record[field]:
+                    record[field] = self.convert_date(record[field], format)
+
+        return record
+
+    def get_records(self, context: dict = None):
+        """Return a generator of record-type dictionary objects."""
+        line_id = 0
+        starting_timestamp = self.get_starting_timestamp(context)
+        start_date = starting_timestamp if starting_timestamp else self.config.get('start_date', '2024-04-01')
+        start_date = (start_date + self.DATE_INCREMENT).strftime(self.DATE_FORMAT)
+        logger.info(f'start_date:{start_date}')
+        try:
+            all_data = self.download_data(start_date, self.api)
+
+            data_io = StringIO(all_data)
+            first_line = data_io.readline().strip()
+            fieldnames = [col.strip().replace(' ', '_').lower() for col in first_line.split('\t')]
+
+            reader = csv.DictReader(data_io, delimiter='\t', fieldnames=fieldnames)
+
+            for record in reader:
+                line_id += 1
+                record['_line_id'] = line_id
+                record['_time_extracted'] = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+                record['_api_report_date'] = start_date
```

Comment:
We are using `_api_report_date` as the `replication_key` because a `start_date` column already exists in the `FinancialReportStream` schema.

Reply:
Ok, we just need to update the state when deploy it to production.